docs/noise.ipynb

```diff
@@ -62,196 +62,673 @@
+    "for prob, kraus in cirq.mixture(bit_phase_flip):\n",
+    "    print(f\"With probability {prob}, apply\\n\", kraus, end=\"\\n\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "72486a155131"
+   },
+   "source": [
+    "Note: Since `_mixture_` is defined, the `cirq.channel` protocol can also be used."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ada775055c3f"
+   },
+   "source": [
+    "The custom channel can be used in a circuit just like other predefined channels."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "1af5c9f60cab"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"Example of using a custom channel in a circuit.\"\"\"\n",
+    "circuit = cirq.Circuit(\n",
+    "    bit_phase_flip.on_each(*cirq.LineQubit.range(3))\n",
+    ")\n",
+    "circuit"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "96a696b543f7"
+   },
+   "source": [
+    "## Simulating noisy circuits"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "38dfcb60ef79"
+   },
+   "source": [
+    "The `cirq.DensityMatrixSimulator` can simulate any noisy circuit (i.e., can apply any quantum channel) because it stores the full density matrix $\\rho$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "a2973a8faef0"
+   },
+   "outputs": [],
+   "source": [
+    "\"\"\"Simulating a circuit with the density matrix simulator.\"\"\"\n",
+    "# Get a circuit.\n",
+    "qbit = cirq.GridQubit(0, 0)\n",
+    "circuit = cirq.Circuit(\n",
+    "    cirq.X(qbit),\n",
+    "    cirq.amplitude_damp(0.1).on(qbit)\n",
+    ")\n",
+    "\n",
+    "# Display it.\n",
+    "print(\"Simulating circuit:\")\n",
+    "print(circuit)\n",
+    "\n",
+    "# Simulate with the density matrix simulator.\n",
+    "dsim = cirq.DensityMatrixSimulator()\n",
+    "rho = dsim.simulate(circuit).final_density_matrix\n",
+    "\n",
+    "# Display the final density matrix.\n",
+    "print(\"\\nFinal density matrix:\")\n",
+    "print(rho)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ac8a991a426b"
+   },
+   "source": [
+    "Note that the density matrix simulator supports the `run` method which only gives access to measurements as well as the `simulate` method (used above) which gives access to the full density matrix."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "6f3caf9cea92"
+   },
+   "source": [
+    "Circuits in which all channels support the `cirq.mixture` protocol can also be simulated with the `cirq.Simulator`. When simulating such a channel, a single Kraus operator is randomly sampled from the mixture (according to the probability distribution) and applied to the wavefunction. This method is known as \"Monte Carlo (wavefunction) simulation\" or \"quantum trajectories.\""
```

Comment:
We do support monte carlo for arbitrary channels not just mixtures: https://github.com/quantumlib/Cirq/pull/3043/files

Reply:
Ahah! To check: Monte Carlo with only channels (not mixtures) will be slower than Monte Carlo with all mixtures because the probability to apply a Kraus depends on the state - is this an accurate remark to include?

Reply:
I didn't know they were both supported though, that's really cool

Reply:
Yes your understanding is correct!